src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -115,13 +115,18 @@ public HashSet(IEnumerable<T> collection, IEqualityComparer<T> comparer)
             var otherAsHashSet = collection as HashSet<T>;
             if (otherAsHashSet != null && AreEqualityComparersEqual(this, otherAsHashSet))
             {
+                if (otherAsHashSet._buckets == null)
+                {
+                    // Source hasn't initialized buckets yet, so neither need this.
+                    Debug.Assert(otherAsHashSet._slots == null);
+                    return;
+                }
                 _buckets = (int[])otherAsHashSet._buckets.Clone();
                 _slots = (Slot[])otherAsHashSet._slots.Clone();
 
                 _count = otherAsHashSet._count;
                 _lastIndex = otherAsHashSet._lastIndex;
                 _freeList = otherAsHashSet._freeList;
-                _version = otherAsHashSet._version;
```

Comment:
Noticed this unnecessary assignment while working on this constructor. Might as well remove it here.